mteb/evaluation/evaluators/ZeroshotClassificationEvaluator.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from sklearn import metrics
+
+from mteb.encoder_interface import Encoder
+from PIL import Image
+from .Evaluator import Evaluator
+
+logger = logging.getLogger(__name__)
+
+
+class ZeroshotClassificationEvaluator(Evaluator):
+    def __init__(
+        self,
+        images:list[Image.Image],
+        labels:list[int],
+        text_candidates:list[str],
+        task_name: str | None = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.images = images
+        self.labels = labels
+        self.text_candidates = text_candidates
+        self.task_name = task_name
+
+    def __call__(self, model: Encoder, *, encode_kwargs: dict[str, Any] = {}):
+        if "batch_size" not in encode_kwargs:
+            encode_kwargs["batch_size"] = 32
+
+        text_embeddings = model.get_text_embeddings(self.text_candidates, batch_size=encode_kwargs["batch_size"])
```

Comment:
we should define these required methods in an  ImageEncoder Protocol (see Encoder) 

One point here is that the ZeroShotClassificationEvaluator is not general enough it be able to take in text-text or image-image. Might be worth it to rename it to: 
ZeroShotImageClassificationEvaluator (same with the abstask)